common/src/main/java/com/redhat/cloud/notifications/models/validation/ValidNonPrivateUrlValidator.java

```diff
@@ -0,0 +1,99 @@
+package com.redhat.cloud.notifications.models.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+public class ValidNonPrivateUrlValidator implements ConstraintValidator<ValidNonPrivateUrl, String> {
+
+    // Error messages.
+    public static final String INVALID_SCHEME = "The provided URL does not contain a valid scheme. It should be either \"http\" or \"https\"";
+    public static final String INVALID_URI = "The provided URL cannot be converted to a valid URI";
+    public static final String INVALID_URL = "The provided URL is invalid";
+    public static final String PRIVATE_IP = "The provided URL's hostname resolves to a private IP";
+    public static final String UNKNOWN_HOST = "The IP address of the provided host cannot be determined";
```

Comment:
These messages may be forwarded to the frontend and showed to the customers (org admins) soon. We could make them a bit more user-friendly:
```suggestion
    public static final String INVALID_SCHEME = "The endpoint URL must start with \"http\" or \"https\"";
    public static final String INVALID_URL = "The endpoint URL is invalid";
    public static final String PRIVATE_IP = "The endpoint URL's host resolves to a private IP";
    public static final String UNKNOWN_HOST = "The IP address of the endpoint URL's host cannot be determined";
```
From a customer perspective, there's no real difference between `INVALID_URI` and `INVALID_URL` which is why I removed `INVALID_URI` from my suggestion.

Reply:
Done!